src/tags/views.py

```diff
@@ -33,48 +38,57 @@
 
 
 class TagList(NonPublicOnlyViewMixin, LoginRequiredMixin, ListView):
-    template_name = 'tags/list.html'
     model = Tag
+    template_name = 'tags/list.html'
 
 
 class TagDetail(NonPublicOnlyViewMixin, LoginRequiredMixin, DetailView):
-    template_name = 'tags/detail.html'
     model = Tag
+    template_name = 'tags/detail.html'
 
     def get_context_data(self, **kwargs):
-        kwargs['quests'] = Quest.objects.filter(tags__name=self.object.name)
-        kwargs['badges'] = Badge.objects.filter(tags__name=self.object.name)
+        ctx = super().get_context_data(**kwargs).copy()
+        ctx['quests'] = Quest.objects.filter(tags__id=self.object.id)
+        ctx['badges'] = Badge.objects.filter(tags__id=self.object.id)
+        return ctx
 
-        return super().get_context_data(**kwargs)
```

Comment:
How come you changed the structure of this?

Reply:
i actually have no idea, reverted